server/src/infra/repositories/activity.repository.ts

```diff
@@ -12,12 +13,13 @@ export interface ActivitySearch {
   isLiked?: boolean;
 }
 
+@DecorateAll(ExecutionTimeHistogram())
 @Injectable()
 export class ActivityRepository implements IActivityRepository {
   constructor(@InjectRepository(ActivityEntity) private repository: Repository<ActivityEntity>) {}
 
   @GenerateSql({ params: [{ albumId: DummyValue.UUID }] })
-  search(options: ActivitySearch): Promise<ActivityEntity[]> {
+  async search(options: ActivitySearch): Promise<ActivityEntity[]> {
```

Comment:
I really dislike the fact that we rely on `async` to make the decorator work. I would much rather just check the return type in the decorator to infer if it is async or not.